### PR TITLE
Localisation_Engine: Added convert methods for Energy and Mass

### LINKED ISSUE #8: Localisation_Toolkit: Add mass conversions
#### Description:

Add the following methods:
- [ ] FromCentigram
- [ ] ToCentigram
- [ ] FromDecagram
- [ ] ToDecagram
- [ ] FromDecigram
- [ ] ToDecigram
- [ ] FromEarthMass
- [ ] ToEarthMass
- [ ] FromGrain
- [ ] ToGrain
- [ ] FromGram
- [ ] ToGram
- [ ] FromHectogram
- [ ] ToHectogram
- [ ] FromKilogram
- [ ] ToKilogram
- [ ] FromKilopound
- [ ] ToKilopound
- [ ] FromKilotonne
- [ ] ToKilotonne
- [ ] FromLongHundredweight
- [ ] ToLongHundredweight
- [ ] FromLongTon
- [ ] ToLongTon
- [ ] FromMegapound
- [ ] ToMegapound
- [ ] FromMegatonne
- [ ] ToMegatonne
- [ ] FromMicrogram
- [ ] ToMicrogram
- [ ] FromMilligram
- [ ] ToMilligram
- [ ] FromNanogram
- [ ] ToNanogram
- [ ] FromOunce
- [ ] ToOunce
- [x] FromPound
- [x] ToPound
- [ ] FromShortHundredweight
- [ ] ToShortHundredweight
- [ ] FromShortTon
- [ ] ToShortTon
- [ ] FromSlug
- [ ] ToSlug
- [ ] FromSolarMass
- [ ] ToSolarMass
- [ ] FromStone
- [ ] ToStone
- [ ] FromTonne
- [ ] ToTonne

### LINKED ISSUE #50: Add Energy conversions
#### Description:

Full list available here - https://github.com/angularsen/UnitsNet/blob/master/Common/UnitDefinitions/Energy.json

### PR DESCRIPTION
**Pull request by Tom-Kingstone**
### NOTE: Depends on 

 ### Issues addressed by this PR

 Closes #8 
 Closes #50 

 
Added methods to convert between Energy units (SI unit: Joules), and between Mass units (SI unit: Kilograms).
Also added respective enums for their units, and some more specific convert methods.

 ### Test files

 ### Changelog

 ### Additional comments

### CODE REVIEW

**Comment by Tom-Kingstone** [comment]
@BHoMBot check copyright-compliance

**Comment by Tom-Kingstone** [comment]
@BHoMBot fix copyright headers ref. 21774941735

**Comment by Tom-Kingstone** [comment]
@BHoMBot check core
@BHoMBot check compliance

**Comment by Tom-Kingstone** [comment]
@BHoMBot check compliance

File: Units_Engine/Convert/Energy/BritishThermalUnit.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+using BH.oM.Base.Attributes;
+using BH.oM.Units;
+using BH.Engine.Base;
+using BH.oM.Quantities.Attributes;
+
+namespace BH.Engine.Units
+{
+    public static partial class Convert
+    {
+        [Description("Convert SI units (joules) into british thermal units")]
```
Comment (FraserGreenroyd) [inline Units_Engine/Convert/Energy/BritishThermalUnit.cs:42]:
```suggestion
        [Description("Convert SI units (joules) into British thermal units")]
```

File: Units_Engine/Convert/Energy/BritishThermalUnit.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        [Input("joules", "The number of joules to convert", typeof(Energy))]
+        [Output("britishThermalUnits", "The number of british thermal units")]
+        public static double ToBritishThermalUnit(this double joules)
+        {
+            UN.QuantityValue qv = joules;
+            return UN.UnitConverter.Convert(qv, EnergyUnit.Joule, EnergyUnit.BritishThermalUnit);
+        }
+
+        [Description("Convert british thermal units into SI units (joules)")]
```
Comment (FraserGreenroyd) [inline Units_Engine/Convert/Energy/BritishThermalUnit.cs:51]:
```suggestion
        [Description("Convert British thermal units into SI units (joules)")]
```

File: Units_Engine/Convert/Energy/Energy.cs [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+namespace BH.Engine.Units
+{
+    public static partial class Convert
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Convert a angle into SI units (joule).")]
```
Comment (FraserGreenroyd) [inline Units_Engine/Convert/Energy/Energy.cs:45]: Description needs updating as we're not converting an angle here.

File: Units_Engine/Convert/Energy/Energy.cs [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+                default:
+                    return null;
+            }
+        }
+    }
+}
+
+
+
+
```
Comment (FraserGreenroyd) [inline Units_Engine/Convert/Energy/Energy.cs:150]:
```suggestion
```

File: Units_Engine/Convert/Energy/MegabritishThermalUnit.cs [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+using BH.oM.Base.Attributes;
+using BH.oM.Units;
+using BH.Engine.Base;
+using BH.oM.Quantities.Attributes;
+
+namespace BH.Engine.Units
+{
+    public static partial class Convert
+    {
+        [Description("Convert SI units (joules) into mega british thermal units")]
```
Comment (FraserGreenroyd) [inline Units_Engine/Convert/Energy/MegabritishThermalUnit.cs:42]:
```suggestion
        [Description("Convert SI units (joules) into mega British thermal units")]
```

File: Units_Engine/Convert/Energy/MegabritishThermalUnit.cs [hunk 6]
```diff
@@ -0,0 +1,61 @@
+/*
```
Comment (FraserGreenroyd) [inline Units_Engine/Convert/Energy/MegabritishThermalUnit.cs:1]: File name needs a capital B on British

File: Units_Engine/Convert/Energy/MegabritishThermalUnit.cs [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        [Description("Convert SI units (joules) into mega british thermal units")]
+        [Input("joules", "The number of joules to convert", typeof(Energy))]
+        [Output("megaBritishThermalUnits", "The number of mega british thermal units")]
+        public static double ToMegabritishThermalUnit(this double joules)
+        {
+            UN.QuantityValue qv = joules;
+            return UN.UnitConverter.Convert(qv, EnergyUnit.Joule, EnergyUnit.MegabritishThermalUnit);
+        }
+
+        [Description("Convert mega british thermal units into SI units (joules)")]
```
Comment (FraserGreenroyd) [inline Units_Engine/Convert/Energy/MegabritishThermalUnit.cs:51]:
```suggestion
        [Description("Convert mega British thermal units into SI units (joules)")]
```

File: Units_Engine/Convert/Mass/Mass.cs [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+                default:
+                    return null;
+            }
+        }
+    }
+}
+
+
+
+
```
Comment (FraserGreenroyd) [inline Units_Engine/Convert/Mass/Mass.cs:147]:
```suggestion
```

File: Units_oM/Enums/Energy.cs [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+using System.Text;
+
+namespace BH.oM.Units
+{
+    public enum EnergyUnit
+    {
+        Undefined,
+        BritishThermalUnit,
+        Joule,
+        KilobritishThermalUnit,
```
Comment (FraserGreenroyd) [inline Units_oM/Enums/Energy.cs:34]:
```suggestion
        KiloBritishThermalUnit,
```

File: Units_oM/Enums/Energy.cs [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+        Kilojoule,
+        KilowattHour,
+        MegabritishThermalUnit,
```
Comment (FraserGreenroyd) [inline Units_oM/Enums/Energy.cs:37]:
```suggestion
        MegaBritishThermalUnit,
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

File: Units_Engine/Convert/Energy/BritishThermalUnit.cs [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+using BH.oM.Quantities.Attributes;
+using UnitsNet.Units;
+
+namespace BH.Engine.Units
+{
+    public static partial class Convert
+    {
+        [Description("Convert SI units (joules) into British thermal units")]
+        [Input("joules", "The number of joules to convert", typeof(Energy))]
+        [Output("britishThermalUnits", "The number of british thermal units")]
```
Comment (FraserGreenroyd) [inline Units_Engine/Convert/Energy/BritishThermalUnit.cs:43]:
```suggestion
        [Output("britishThermalUnits", "The number of British thermal units")]
```

File: Units_Engine/Convert/Energy/BritishThermalUnit.cs [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+        public static double ToBritishThermalUnit(this double joules)
+        {
+            UN.QuantityValue qv = joules;
+            return UN.UnitConverter.Convert(qv, EnergyUnit.Joule, EnergyUnit.BritishThermalUnit);
+        }
+
+        [Description("Convert British thermal units into SI units (joules)")]
+        [Input("britishThermalUnits", "The number of british thermal units to convert")]
```
Comment (FraserGreenroyd) [inline Units_Engine/Convert/Energy/BritishThermalUnit.cs:51]:
```suggestion
        [Input("britishThermalUnits", "The number of British thermal units to convert")]
```

File: Units_Engine/Convert/Energy/MegaBritishThermalUnit.cs [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+using BH.Engine.Base;
+using BH.oM.Quantities.Attributes;
+
+namespace BH.Engine.Units
+{
+    public static partial class Convert
+    {
+        [Description("Convert SI units (joules) into mega British thermal units")]
+        [Input("joules", "The number of joules to convert", typeof(Energy))]
+        [Output("megaBritishThermalUnits", "The number of mega british thermal units")]
```
Comment (FraserGreenroyd) [inline Units_Engine/Convert/Energy/MegaBritishThermalUnit.cs:43]:
```suggestion
        [Output("megaBritishThermalUnits", "The number of mega British thermal units")]
```

File: Units_Engine/Convert/Energy/MegaBritishThermalUnit.cs [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+        public static double ToMegaBritishThermalUnit(this double joules)
+        {
+            UN.QuantityValue qv = joules;
+            return UN.UnitConverter.Convert(qv, EnergyUnit.Joule, EnergyUnit.MegabritishThermalUnit);
+        }
+
+        [Description("Convert mega British thermal units into SI units (joules)")]
+        [Input("megaBritishThermalUnits", "The number of mega british thermal units to convert")]
```
Comment (FraserGreenroyd) [inline Units_Engine/Convert/Energy/MegaBritishThermalUnit.cs:51]:
```suggestion
        [Input("megaBritishThermalUnits", "The number of mega British thermal units to convert")]
```

File: Units_Engine/Convert/Mass/Mass.cs [hunk 15]
```diff
@@ -0,0 +1,144 @@
...
+namespace BH.Engine.Units
+{
+    public static partial class Convert
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Convert a mass into SI units (kiloGram).")]
```
Comment (FraserGreenroyd) [inline Units_Engine/Convert/Mass/Mass.cs:45]:
```suggestion
        [Description("Convert a mass into SI units (kilogram).")]
```

File: Units_Engine/Convert/Mass/Mass.cs [hunk 16]
```diff
@@ -0,0 +1,144 @@
...
+                return UN.UnitConverter.Convert(qv, unUnit, unitSI);
+
+            Compute.RecordError("Unit was undefined. Please use the appropriate BHoM Units Enum.");
+            return double.NaN;
+        }
+
+        /***************************************************/
+
+        [Description("Convert SI units (kilogram) into another mass unit.")]
+        [Input("kilogram", "The number of kiloGram to convert.")]
```
Comment (FraserGreenroyd) [inline Units_Engine/Convert/Mass/Mass.cs:71]:
```suggestion
        [Input("kilogram", "The number of kilogram to convert.")]
```

File: Units_Engine/Convert/Mass/Mass.cs [hunk 17]
```diff
@@ -0,0 +1,144 @@
...
+            UN.QuantityValue qv = kilogram;
+            UNU.MassUnit unitSI = UNU.MassUnit.Kilogram;
+            UNU.MassUnit? unUnit = ToMassUnit(unit);
+
+            if (unUnit != null)
+                return UN.UnitConverter.Convert(qv, unitSI, unUnit);
+
+            Compute.RecordError("Unit was undefined. Please use the appropriate BHoM Units Enum.");
+            return double.NaN;
+
```
Comment (FraserGreenroyd) [inline Units_Engine/Convert/Mass/Mass.cs:91]:
```suggestion
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
Nearly there @Tom-Kingstone 😉 

Also, we could do with some unit tests for the new methods please, otherwise @peterjamesnugent will be unhappy 😄

**Comment by FraserGreenroyd** [comment]
@BHoMBot check unit-tests

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by Tom-Kingstone** [comment]
@BHoMBot check versioning
@BHoMBot check installer

**Comment by Tom-Kingstone** [comment]
@BHoMBot check serialisation
@BHoMBot check null-handling

**Review by FraserGreenroyd — APPROVED** [review]
Everything looks good now on review, thanks @Tom-Kingstone

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**